From 95c4523085f1345f1c2e3557744acc6d7698a882 Mon Sep 17 00:00:00 2001
From: chenzhanzhan <chenzhanzhan@eswincomputing.com>
Date: Fri, 20 Jun 2025 09:38:03 +0800
Subject: [PATCH 577/652] feature: Fix kexec job depend on proc/iomem reserved

Changelogs:
1. Fix kexec job depend on proc/iomem reserved(Modification of the kernel source code)

Signed-off-by: chenzhanzhan <chenzhanzhan@eswincomputing.com>
---
 arch/riscv/mm/init.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/riscv/mm/init.c b/arch/riscv/mm/init.c
index 6584317decd5..3d783bfb05ed 100644
--- a/arch/riscv/mm/init.c
+++ b/arch/riscv/mm/init.c
@@ -1511,6 +1511,8 @@ static void __init reserve_crashkernel(void)
 
 	crashk_res.start = crash_base;
 	crashk_res.end = crash_base + crash_size - 1;
+	pr_info("insert_resource:crashk_res(0x%lx, 0x%lx)\n", crashk_res.start, crashk_res.end);
+	insert_resource(&iomem_resource, &crashk_res);
 }
 
 void __init paging_init(void)
-- 
2.52.0

